### PR TITLE
Admin interface locale breaks  "Expiration date" field

### LINKED ISSUE #32497: Admin interface locale breaks  "Expiration date" field
### Preconditions (*)
1. Magento version 2.4.2

### Steps to reproduce (*)
1. sigh in to admin panel
1. go to System -> All users
1. change locale of current user to Any except 'en_Us'
1. try to create a new user, and set expiration date 
1. save user
1. get an error "Expiration Date is not valid"

### Expected result (*)
1. expect to create user

### Actual result (*)
1. !https://user-images.githubusercontent.com/33631491/111330717-df91bd80-8678-11eb-91c4-824bccdf455c.jpeg!

## 
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- < > Severity: **S2** *- Affects non-critical data or functionality and forces users to employ a workaround.*

@magento give me 2.4-develop instance

### PR DESCRIPTION
**Pull request by engcom-Kilo**
### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/magento2/issues/32497

### Manual testing scenarios (*)

1. see https://github.com/magento/magento2/issues/32497

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Kilo** [comment]
@magento run all tests

File: app/code/Magento/Security/Model/UserExpiration/Validator.php [hunk 1]
```diff
@@ -16,16 +16,12 @@
...
 {
-
-    /**@var TimezoneInterface */
+    /**
+     * @var TimezoneInterface
+     */
     private $timezone;
 
-    /**@var DateTime */
-    private $dateTime;
```
Comment (nuzil) [inline app/code/Magento/Security/Model/UserExpiration/Validator.php:24]: Please also remove class from "use" section

**Review by nuzil — CHANGES_REQUESTED** [review]

**Comment by nuzil** [comment]
@magento run all tests

**Comment by nuzil** [comment]
@magento give me test instance

**Comment by nuzil** [comment]
Hey @engcom-Kilo  I still see this error on test instance

File: app/code/Magento/Security/Test/Unit/Model/UserExpiration/ValidatorTest.php [hunk 2]
```diff
@@ -18,37 +19,35 @@
...
-     * @var MockObject|\Magento\Framework\Stdlib\DateTime\DateTime
+     * @var MockObject|TimezoneInterface
      */
-    private $dateTimeMock;
-
-    /**@var \PHPUnit\Framework\MockObject\MockObject|\Magento\Framework\Stdlib\DateTime\TimezoneInterface */
     private $timezoneMock;
 
+    /**
+     * @inheirtDoc
```
Comment (zakdma) [inline app/code/Magento/Security/Test/Unit/Model/UserExpiration/ValidatorTest.php:33]: @inheritDoc

File: dev/tests/integration/testsuite/Magento/Security/Model/UserExpiration/ValidatorTest.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+     */
+    private $localeResolver;
+
+    /**
+     * @var TimezoneInterface
+     */
+    private $timeZone;
+
+    /**
+     * @inheridoc
```
Comment (zakdma) [inline dev/tests/integration/testsuite/Magento/Security/Model/UserExpiration/ValidatorTest.php:36]: @inheritDoc

**Review by zakdma — CHANGES_REQUESTED** [review]

**Pull request closed without merging**